corpus/proza/parse.py

```diff
@@ -0,0 +1,187 @@
+            collection_name = collection_name_w_cap[:-len(cnt_substr) - 1]
+            collections.append((collection_name, collection_publications_cnt, href))
+    return collections
+
+
+def get_books_from_collection(collection_url, collection_name: CollectionName) -> List[Book]:
+    books = []
+    html, soup = http_client.get_html(BASE_URL + f"/{collection_url}")
+    for y in soup.find_all("div", {"id": "bookheader"}):
+        found_collection = y.find(text=collection_name)
+        if found_collection:
+            book_list = y.next_sibling.next_sibling
+            children = book_list.findChildren()
+            for book_tag in children:
+                a = book_tag.find('a')
+                if a:
+                    small_text = book_tag.find('small').text
+                    publication_date_time_str = small_text.split(',')[1].lstrip().rstrip()
+                    category = small_text.split(',')[0].lstrip().rstrip()[2:]
+                    if category != 'фэнтези': return []
+                    try:
+                        dt = datetime.strptime(publication_date_time_str, '%d.%m.%Y %H:%M')
+                    except ValueError:
+                        dt = datetime.strptime(publication_date_time_str, '%d.%m.%Y')
+                    books.append(Book(a.text, a.get("href"), dt))
+    return sorted(books, key=lambda b: b.publication_date)
+
+
+def get_book_content(book: Book):
+    review_cnt = 0
+    html, soup = http_client.get_html(BASE_URL + book.rel_url)
+    notes_tag = soup.find(text="На это произведение написаны ")
+    # notes_tag = soup.find("div", {"class": "notesline", "text": "На это произведение написаны "})
+    if notes_tag:
+        review_cnt = int(notes_tag.parent.find('b').text.split()[0])
+    try:
+        t = soup.find("div", {"class": "text"}).text.replace('\xa0', ' ')
+        return t, review_cnt
+    except AttributeError:
+        return "", review_cnt
+
+
+def dump_if_large(books, col_url, col_name, author_url, author_info_by_url) -> bool:
+    # книги из одной коллекции
+    books.sort(key=lambda b: b.publication_date)
+    # print(books)
+    f_name = mddumper.get_file_name(author_url, col_name, MDSTORAGE) + ".md"
+    if os.path.isfile(f_name):
+        # print(f"skip {f_name}")
+        return True
+    for b in books:
+        b.content, b.review_cnt = get_book_content(b)
+        if len(b.content) < REQUIRED_BOOK_SIZE: return False
+    total_length = sum(len(b.content) for b in books)
+    if total_length > REQUIRED_SIZE:
+        mddumper.dump(books, col_name, col_url, author_url, total_length, author_info_by_url[author_url])
+        print(f'dumped {f_name}')
+        return True
+    # print(f'candidate too small {BASE_URL + col_url}. total_size = {total_length}')
+    return False
+
+
+seen_authors = set()
+author_info_by_url = {}
+for year, month, day in tqdm(list(itertools.product(YEARS, MONTHS, DAYS)), ncols=80, desc="DAYS"):
```

Comment:
[START_DATE+timedelta(days=x) for x in range(1000)]

Reply:
fixed